src/sema/using.rs

```diff
@@ -251,7 +264,22 @@ pub(crate) fn using_decl(
                         Some(oper)
                     } else {
                         if let Some(ty) = &ty {
-                            if *ty != func.params[0].ty {
+                            let dummy = Expression::Variable {
+                                loc,
+                                ty: ty.clone(),
+                                var_no: 0,
+                            };
+
+                            if dummy
+                                .cast(
+                                    &loc,
+                                    &func.params[0].ty,
+                                    true,
+                                    ns,
+                                    &mut Diagnostics::default(),
+                                )
+                                .is_err()
```

Comment:
Is there any test in which we are casting the first argument for the library call?

Reply:
The polkadot test does this